scipy/signal/filter_design.py

```diff
@@ -3557,6 +3557,8 @@ def buttord(wp, ws, gpass, gstop, analog=False, fs=None):
     elif filter_type == 4:          # pass
         nat = ((stopb ** 2 - passb[0] * passb[1]) /
                (stopb * (passb[0] - passb[1])))
+    else:
+        raise ValueError("Bad filter type: %s" % filter_type)
```

Comment:
Perhaps worth noting that this code path is never hit in the test suite, for this or any of the other similar `ValueError` entries adjusted in this PR.

Reply:
That's a good point

Reply:
Probably worth adding some `with pytest.raises(ValueError, match='Bad filter type'):` to the tests